loopout_test.go

```diff
@@ -15,6 +15,105 @@ import (
 	"github.com/lightninglabs/loop/test"
 )
 
+// TestLoopOutPayment tests the first part of the loop out process up to the
+// point where the off-chain payments are made.
+func TestLoopOutPayment(t *testing.T) {
+	defer test.Guard(t)()
+
+	// Set up test context objects.
+	lnd := test.NewMockLnd()
+	ctx := test.NewContext(t, lnd)
+	server := newServerMock()
+	store := newStoreMock(t)
+
+	expiryChan := make(chan time.Time)
+	timerFactory := func(_ time.Duration) <-chan time.Time {
+		return expiryChan
+	}
+
+	height := int32(600)
+
+	cfg := &swapConfig{
+		lnd:    &lnd.LndServices,
+		store:  store,
+		server: server,
+	}
+
+	sweeper := &sweep.Sweeper{Lnd: &lnd.LndServices}
+
+	blockEpochChan := make(chan interface{})
+	statusChan := make(chan SwapInfo)
+
+	const maxParts = 5
```

Comment:
how about extending this test just a little bit, such that max parts is a parameter and we test that with zero it fails, and then 1, 2, say 5?

Reply:
We can't just test that at this level. There is no real lnd or pathfinding involved. The test just asserts passing through the parameter. This more suitable for an itest, although it partially redoes the lnd integration test. In loop we can assume the lnd part of mpp to be working.

Reply:
Thanks for explaining, I missed that. Maybe if LND mock could be parametrized then we could still test our side a bit better. I agree it is much more convenient with itest though.